scripts/run_messaging_hub.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+
+import argparse
+# import multiprocessing
+import sys
+import threading
+import time
+
+from pocs.utils.config import load_config
+from pocs.utils.logger import get_root_logger
+from pocs.utils.messaging import PanMessaging
+
+
+def run_forwarder(logger, sub_port, pub_port):
+    msg = 'Starting {} -> {} forwarder'.format(sub_port, pub_port)
+    print(msg)
+    logger.info(msg)
+    try:
+        PanMessaging.create_forwarder(sub_port, pub_port)
+    except Exception:
+        pass
+    msg = 'Forwarder for {} -> {} has stopped'.format(sub_port, pub_port)
+    print(msg)
+    logger.info(msg)
+
+
+def run_forwarders(sub_and_pub_pairs):
+    logger = get_root_logger()
+    logger.info('Starting forwarders')
+    threads = []
+    for sub, pub in sub_and_pub_pairs:
+        name = 'fwd_{}_to_{}'.format(sub, pub)
+        t = threading.Thread(target=run_forwarder, name=name, args=(logger, sub, pub), daemon=True)
+        logger.info('Starting thread {}', name)
+        t.start()
+        threads.append(t)
+        time.sleep(0.05)
+    time.sleep(0.2)
+    if not any([t.is_alive() for t in threads]):
+        msg = 'Failed to start any forwarder!'
+        logger.error(msg)
+        print(msg)
+    else:
+        print()
+        print('Hit Ctrl-c to stop')
+    try:
+        # Keep running until they've all died.
+        while threads:
+            for t in threads:
+                t.join(timeout=100)
+                if t.is_alive():
+                    continue
+                logger.info('Thread {} has stopped', t.name)
+                threads.remove(t)
+                break
+        # If we get here, then the forwarders died for some reason.
+        sys.exit(1)
+    except KeyboardInterrupt:
+        sys.exit(0)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description='Record sensor data from an Arduino and send it relay commands.')
```

Comment:
Change description here.

Reply:
Is something very wrong with that? I don't see it.

Reply:
I don't think the messenger hub (this script) is necessarily related to just the arduino and sensor data. It handles the POCS `say` messages and shutdown commands, etc. It's more generic than just the Arduinos. Or is that not true?

In my head the description for this script would be something like "Sets up a publish and subscribe messaging hub for various communication between various components of the system" (don't feel tied to that text at all).

Reply:
Sorry, I'm an idiot. I got confused between this script and another that is related to the sensors, from which I started when creating this new file. I'll send a fix.